frontend/src/__tests__/cypress/cypress/e2e/pipelines/PipelineRuns.cy.ts

```diff
@@ -170,6 +171,92 @@ describe('Pipeline runs', () => {
       });
     });
 
+    describe('table pagination', () => {
+      it('Active run table pagination', () => {
+        const mockRuns = Array.from({ length: 15 }, (_, i) =>
+          buildMockRunKF({
+            display_name: `Test active run-${i}`,
+            run_id: `run-${i}`,
+            pipeline_version_reference: {
+              pipeline_id: pipelineId,
+              pipeline_version_id: `test-version-${i}`,
+            },
+            experiment_id: `test-experiment-${i}`,
+            created_at: '2024-02-05T00:00:00Z',
+            state: RuntimeStateKF.SUCCEEDED,
+          }),
+        );
+
+        cy.intercept(
+          {
+            method: 'GET',
+            pathname: `/api/service/pipelines/${projectName}/dspa/apis/v2beta1/runs`,
+          },
+          {
+            runs: mockRuns.slice(0, 10),
+            total_size: 15,
+            next_page_token: 'page-2-token',
+          },
+        );
```

Comment:
Pagination relies on the UI sending the correct query params in the network request as the user navigates the pages. The test should assert the query params are correct, especially the page token as it controls the ability to retrieve the next page. Page tokens are cached in the UI and therefore going back a page should use the cached value. Any new page token received from the backend should be used in subsequent next page requests and not the old cached value.

Interceptors should be configured not only with a path, but also query params.